File: DTOs/ReservationDto.cs
public class ReservationDto
{
    public int Id { get; set; }

    public DateTime StartTime { get; set; }
    public DateTime PlannedEndTime { get; set; }

    public double InitialAmount { get; set; }
    public double ExtraAmount { get; set; }
    public double TotalAmount { get; set; }

    public string Status { get; set; } = string.Empty;

    public string ParkingName { get; set; } = string.Empty;
    public string ZoneName { get; set; } = string.Empty;
    public string SpotCode { get; set; } = string.Empty;
}

File: frontend/smart-parking/src/app/models/reservation.dto.ts
export enum ReservationStatus {
  Created   = 'Created',
  Active    = 'Active',
  Completed = 'Completed',
  Cancelled = 'Cancelled',
}
 
export interface CreateReservationDto {
  spotId: number;
  startTime: string;
}
 
/** Returned by GET /api/reservations — status is a string (.ToString()) */
export interface Reservation {
  id: number;
  startTime: string;
  plannedEndTime: string;
  initialAmount: number;
  extraAmount: number;
  totalAmount: number;
  status: 'Created' | 'Active' | 'Completed' | 'Cancelled';
  parkingName: string;
  zoneName: string;
  spotCode: string;
}
 
/**
 * Returned by POST /api/reservations
 * Renamed from PaymentInitResponse → BookingSessionResponse
 * to avoid clashing with the DOM's built-in PaymentResponse global.
 */
export interface BookingSessionResponse {
  message: string;
  paymentSessionId: number;
  paymentUrl: string;
  amount: number;
  startTime: string;
  expiresAt: string;
}